<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1217</width>
    <height>773</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QLabel" name="project_label">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>0</y>
      <width>351</width>
      <height>71</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>30</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>OS Scheduler</string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>32</x>
      <y>92</y>
      <width>581</width>
      <height>141</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item alignment="Qt::AlignLeft">
        <widget class="QLabel" name="algo_label">
         <property name="font">
          <font>
           <pointsize>15</pointsize>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Scheduling Algorithm</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="algorithm_comboBox">
         <property name="iconSize">
          <size>
           <width>16</width>
           <height>16</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string>1. FCFS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2. SJF</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>3. Priority</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4. Round Robin</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <item>
        <widget class="QLabel" name="no_process_label">
         <property name="font">
          <font>
           <pointsize>15</pointsize>
           <bold>true</bold>
          </font>
         </property>
         <property name="text">
          <string>Number of Process</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="no_of_process_value"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>32</x>
      <y>240</y>
      <width>581</width>
      <height>37</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="algo_type">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="font">
        <font>
         <pointsize>15</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Type</string>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <property name="spacing">
        <number>8</number>
       </property>
       <item>
        <widget class="QRadioButton" name="Non_preemptive">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Non Preemptive</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QRadioButton" name="preemptive">
         <property name="font">
          <font>
           <pointsize>12</pointsize>
           <kerning>true</kerning>
          </font>
         </property>
         <property name="text">
          <string>Preemptive</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>280</y>
      <width>581</width>
      <height>35</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <widget class="QLabel" name="quantum_label">
       <property name="font">
        <font>
         <pointsize>15</pointsize>
         <bold>true</bold>
        </font>
       </property>
       <property name="text">
        <string>Quantum</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="quantum_value"/>
     </item>
    </layout>
   </widget>
   <widget class="QTableWidget" name="data_table">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>340</y>
      <width>421</width>
      <height>291</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="simulate_button">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>480</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Simulate</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="gantt_chart">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>660</y>
      <width>1141</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="avg_wait">
    <property name="geometry">
     <rect>
      <x>730</x>
      <y>550</y>
      <width>401</width>
      <height>61</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>12</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true"> border-style: solid;
  border-color: black;
  border-width: 7px;</string>
    </property>
    <property name="text">
     <string>Average Waiting Time : </string>
    </property>
   </widget>
   <widget class="QWidget" name="layoutWidget">
    <property name="geometry">
     <rect>
      <x>750</x>
      <y>230</y>
      <width>361</width>
      <height>311</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <widget class="QLabel" name="label">
       <property name="font">
        <font>
         <pointsize>12</pointsize>
        </font>
       </property>
       <property name="text">
        <string>Processes  Time Line</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QTableWidget" name="timeline"/>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="LiveGantt">
    <property name="geometry">
     <rect>
      <x>460</x>
      <y>560</y>
      <width>211</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string>Live Gantt Chart</string>
    </property>
   </widget>
   <widget class="QLabel" name="timerLabel">
    <property name="geometry">
     <rect>
      <x>580</x>
      <y>400</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
     </font>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>480</x>
      <y>400</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <pointsize>15</pointsize>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Timer :</string>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>630</x>
      <y>90</y>
      <width>521</width>
      <height>123</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <widget class="QTableWidget" name="data_table_2">
       <property name="rowCount">
        <number>1</number>
       </property>
       <property name="columnCount">
        <number>4</number>
       </property>
       <row/>
       <column>
        <property name="text">
         <string>Process Id</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Burst time</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Arrival time</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>priority</string>
        </property>
       </column>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="AddDynamically">
       <property name="text">
        <string>Add</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
